test/intest4.F90

```diff
@@ -0,0 +1,87 @@
+! This is a test for NCEPLIBS-bufr.
+!
+! Reads test file 'testfiles/IN_4' using CRBMG with OPENBF IO = 'SEC3'
+! using bitmap and marker operators.
+!
+! Ed Hartnett, J. Ator, 2/16/23
+program intest4
+  implicit none
+  integer*4 ireadsb, iupbs01, iupbs3, ibfms
+  integer*4 mxbf, nbyt, ierr
+  integer mxbfd4, mxds3, mxr8lv, mxr8pm
+  integer ier1,ier2, ier3, ierme, imgdt, nds3
+  integer nr8lv, nr8lv2, ntag1, ntag2, ntag3
+  parameter (mxbf = 20000)
+  parameter (mxbfd4 = mxbf/4)
+  parameter (mxds3 = 60)
+  parameter (mxr8pm = 10)
+  parameter (mxr8lv = 255)
+  real*8 r8arr (mxr8pm, mxr8lv), r8arr2 (mxr8pm, mxr8lv)
+  integer ibfmg (mxbfd4)
+  character cmgtag*8, bfmg(mxbf), cds3(mxds3)*6, tag1*8, tag2*8, tag3*8
+  character*20 filnam / 'testfiles/IN_4' /
+  character filost / 'r' /
+  equivalence (bfmg (1), ibfmg (1))
+
+  print *, 'Testing reading IN_1, CRBMG with OPENBF IO = SEC3 ', &
+       'using bitmap and marker operators.'
+
+#ifdef KIND_8
+  call setim8b(.true.)
+#endif
+
+  ! Open the test file.
+  call cobfl(filnam, filost)
+
+  ! Set date format.
+  call datelen(10)
+
+  ! Open /dev/null for output. ???
```

Comment:
Jeff what is going on with opening /dev/null on unit 11? What is it used for?

Reply:
Since for this test we're using cobfl to read in a BUFR message directly from filnam, then we won't actually be reading from the logical unit pointed to via openbf.  However, we still need to call openbf to set io = 'SEC3' (to tell it to decode messages according to Section 3), and since we still need to call openbf, then we still need to pass in a logical unit number that points to some file on the system.   So, bottom line, since in this case we'll never actually try to read anything from that logical unit, then /dev/null works as well as anything and avoids having to otherwise create an empty dummy file that we'll never read from.

I realize this can be confusing for anyone who's not familiar with all of the different ways that BUFR messages can be input to the library, so I'll go ahead and document this in the code.